infra/staging/main.tf

```diff
@@ -203,13 +203,25 @@ resource "google_compute_managed_ssl_certificate" "call_home_frontend_lb_ssl" {
   managed {
     domains = ["app2-staging.callhome.sg"]
   }
+
+  lifecycle {
+    create_before_destroy = true
+  }
 }
 
 // HTTP proxy
-resource "google_compute_target_http_proxy" "call_home_frontend_lb_proxy" {
+resource "google_compute_target_https_proxy" "call_home_frontend_lb_proxy" {
   name = "call-home-frontend-lb-proxy"
   project = data.google_project.project.project_id
   url_map = google_compute_url_map.call_home_frontend_lb_url_map.id
+
+  ssl_certificates = [
+    google_compute_managed_ssl_certificate.call_home_frontend_lb_ssl.name
+  ]
+
+  depends_on = [
+    google_compute_managed_ssl_certificate.call_home_frontend_lb_ssl
```

Comment:
Why do we need this explicit `depends_on`? This resource depends on `google_compute_managed_ssl_certificate.call_home_frontend_lb_ssl.name
`, so I'd expect Terraform to already compute this dependency for us.

I see a similar pattern elsewhere. Do take a look at those too.

Reply:
Fixed in https://github.com/bettersg/call-home/pull/145/commits/f2aacfb358f174a88a8554b4996bfde3fa5aeb9f